src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder();
+        LocalDate localDateFrom = parseDateFromString(dateFrom);
+        LocalDate localDateTo = parseDateFromString(dateTo);
+
+        report.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        for (String currentName : names) {
+
+            int salaryAmount = 0;
+            for (String record : data) {
+                if (record == null || record.isEmpty()) {
+                    continue;
+                }
```

Comment:
This check for null or empty records is not necessary. If your data input is correct, there should not be any null or empty records. If there are, it's better to let the program throw an exception so the error can be fixed at the source. Removing this check will make your code cleaner.